### PR TITLE
avoid pulling image while service's rolling update

### PR DESCRIPTION
**Pull request by carmark**

### CODE REVIEW

File: api/client/service.go [hunk 1]
```diff
@@ -319,8 +319,10 @@ func (cli *DockerCli) CmdServiceRolling_update(args ...string) error {
 	}
 
 	ctx := context.Background()
-	if err := cli.pullImage(ctx, *flImage); err != nil {
-		return err
+	if _, _, err := cli.client.ImageInspectWithRaw(ctx, *flImage, false); err != nil {
```
Comment (feiskyer) [inline api/client/service.go]: check the error details? e.g. only pull on image not exist

Comment (carmark) [reply api/client/service.go]: updated.

**Comment by carmark** [comment]
```
[lei@h8s-single hypercli]$ ./hyper/hyper service create --service-port=80 --label=app=nginx --name=http --replicas=3 nginx
Service http is created.
[lei@h8s-single hypercli]$ ./hyper/hyper pull httpd
Using default tag: latest
latest: Pulling from library/httpd
ad74af05f5a2: Pull complete
3d839585b9c7: Pull complete
cf157792586a: Pull complete
c620105f0566: Pull complete
830b826a2e13: Pull complete
ec2eb5743536: Pull complete
eb53f3c09897: Pull complete
Digest: sha256:5b35d13089db73df620f4c198f5a4bfa56b8fe45a0364f343df9a26d874fef6c
Status: Downloaded newer image for httpd:latest

[lei@h8s-single hypercli]$ ./hyper/hyper ps -a
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS                                              NAMES                   PUBLIC IP
cab80ba4f693        nginx               "nginx -g 'daemon off"   About a minute ago   Up About a minute   0.0.0.0:80->80/tcp                                 http-c4r1m1mgbq1ft0mo
c1e0a3c546b4        nginx               "nginx -g 'daemon off"   About a minute ago   Up About a minute   0.0.0.0:80->80/tcp                                 http-n3afcsnohrbsmrh2
4acb5d49e524        nginx               "nginx -g 'daemon off"   About a minute ago   Up About a minute   0.0.0.0:80->80/tcp                                 http-4jhwm55vrgtu91zi
[lei@h8s-single hypercli]$ ./hyper/hyper service rolling-update --image httpd http
Rolling-update is requested for service http.
[lei@h8s-single hypercli]$
```

**Comment by feiskyer** [comment]
LGTM

**Pull request merged**